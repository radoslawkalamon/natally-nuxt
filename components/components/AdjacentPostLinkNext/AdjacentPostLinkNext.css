.adjacent-post-link-next {
  color: inherit;
  display: flex;
  flex-flow: column nowrap;
  gap: var(--gap);
  max-width: var(--max-width-adjacent-post-link);
  text-align: right;
  text-decoration: none;
  width: 100%;
}

.adjacent-post-link-next__label {
  font-size: var(--font-size-footnotes);
  line-height: var(--line-height-footnotes);
}

.adjacent-post-link-next__label::after {
  color: var(--color-accent);
  content: ' ' var(--content-arrows-right);
}

.adjacent-post-link-next__title {
  font-family: var(--font-family-sans);
  font-size: var(--font-size-header-3);
  font-weight: bold;
  line-height: var(--line-height-header);
  text-decoration: underline;
}

.adjacent-post-link-next:hover .adjacent-post-link-next__title {
  text-decoration: none;
}
