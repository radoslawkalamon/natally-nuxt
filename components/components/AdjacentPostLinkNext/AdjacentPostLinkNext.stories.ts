import type { Meta, Story } from '@storybook/vue'
import ComponentsAdjacentPostLinkNext from '@/components/components/AdjacentPostLinkNext/AdjacentPostLinkNext.vue'

const meta: Meta = {
  title: 'Components / Adjacent Post Link Next',
  component: ComponentsAdjacentPostLinkNext
}
export default meta

const Template: Story = (_args, { argTypes }) => ({
  components: {
    ComponentsAdjacentPostLinkNext
  },
  props: Object.keys(argTypes),
  template: '<ComponentsAdjacentPostLinkNext v-bind="$props" />'
})

export const Default: Story = Template.bind({})
Default.args = {
  label: 'NastÄ™pny',
  path: '/post/test-post/',
  title: 'Post testowy'
}
