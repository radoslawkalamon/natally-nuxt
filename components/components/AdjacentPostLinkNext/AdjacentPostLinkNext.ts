import Vue from 'vue'

export default Vue.extend({
  name: 'ComponentsAdjacentPostLinkNext',
  props: {
    label: {
      type: String,
      required: true
    },
    path: {
      type: String,
      required: true
    },
    title: {
      type: String,
      required: true
    }
  }
})
