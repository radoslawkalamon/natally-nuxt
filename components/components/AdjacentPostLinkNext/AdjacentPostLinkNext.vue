<template>
  <NuxtLink
    class="adjacent-post-link-next"
    data-test="components-adjacent-post-link-next"
    :to="path"
  >
    <span
      class="adjacent-post-link-next__title"
      v-text="title"
    />
    <span
      class="adjacent-post-link-next__label"
      v-text="label"
    />
  </NuxtLink>
</template>

<script src="./AdjacentPostLinkNext.ts" lang="ts"></script>
<style src="./AdjacentPostLinkNext.css" scoped></style>
