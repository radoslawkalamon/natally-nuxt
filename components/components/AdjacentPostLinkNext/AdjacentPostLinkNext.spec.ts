import AdjacentPostLinkNext from '@/components/components/AdjacentPostLinkNext/AdjacentPostLinkNext.vue'
import { shallPassUnitSanityTest } from '@/devtools/jest.common.spec'

describe('Components / Adjacent Post Link Next', () => {
  describe('Unit', () => {
    shallPassUnitSanityTest({
      component: AdjacentPostLinkNext,
      options: {
        propsData: {
          label: 'NastÄ™pny',
          path: '/post/test-path/',
          title: 'Post testowy'
        }
      }
    })
  })
})
