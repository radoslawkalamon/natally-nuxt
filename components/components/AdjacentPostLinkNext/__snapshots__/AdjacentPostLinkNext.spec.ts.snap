// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / Adjacent Post Link Next Unit shall pass unit sanity test 1`] = `"<nuxtlink-stub to=\\"/post/test-path/\\" class=\\"adjacent-post-link-next\\"><span class=\\"adjacent-post-link-next__title\\">Post testowy</span> <span class=\\"adjacent-post-link-next__label\\">NastÄ™pny</span></nuxtlink-stub>"`;
